geode-core/src/main/java/org/apache/geode/internal/cache/ExpiryTask.java

```diff
@@ -149,13 +155,34 @@ long getExpiryMillis() throws EntryNotFoundException {
    */
   protected boolean isExpirationPossible() throws EntryNotFoundException {
     long expTime = getExpirationTime();
-    if (expTime > 0L && getNow() >= expTime) {
-      return true;
+    if (expTime > 0L) {
```

Comment:
What should happen if expiration is due to ttl?

Reply:
Then it is okay to just expire based on the local information.
Since last modified time is always changed by a write and since writes go to every copy of the data the ttl check only needs to happen locally.
The way this works in this fix is that the implementation of isIdleExpiredOnOthers
does this check:
    if (getIdleAttributes().getTimeout() <= 0L) {
      // idle expiration is not being used
      return true;
    }
So if you expiration configuration does not care about idle time then we just return true which means expiration is possible.
If we are using idle time then we fetch it from others and redo this calculation which may cause is to still expire due to ttl (note that you can have both idle and ttl configured on a region).

Reply:
@dschneider-pivotal Thanks Darrel. I thought the same that we should be expiring if its TTL.
`getExpirationTime()` call in the previous line also checks if TTL or Idle and returns expiryTime based on the what is configured.
Probably its more readable if the check is part of this function rather than in `isIdleExpiredOnOthers`.